cmake/onnxruntime_providers.cmake

```diff
@@ -198,6 +207,12 @@ if (onnxruntime_ENABLE_TRAINING)
   if (onnxruntime_USE_NCCL OR onnxruntime_USE_MPI)
     target_include_directories(onnxruntime_providers PUBLIC ${MPI_INCLUDE_DIRS})
   endif()
+
+  # DLPack is a header-only dependency
+  set(DLPACK_INCLUDE_DIR ${PROJECT_SOURCE_DIR}/external/dlpack/include)
```

Comment:
as we moved the dlpack to /core

in onnxruntime_python.cmake, do we still need to build with dlpack dependency? 
I see following line is still in python.cmake

set(DLPACK_INCLUDE_DIR ${PROJECT_SOURCE_DIR}/external/dlpack/include)

Reply:
We still need it as "#include "core/dlpack/dlpack_converter.h" will try to include <dlpack/dlpack.h> at the end. It won't compile if remove.